clusters/clusters_api.go

```diff
@@ -361,8 +367,8 @@ type Cluster struct {
 	ClusterName string `json:"cluster_name,omitempty"`
 
 	SparkVersion              string     `json:"spark_version"`
-	NumWorkers                int32      `json:"num_workers" tf:"group:size"`
-	Autoscale                 *AutoScale `json:"autoscale,omitempty" tf:"group:size"`
+	NumWorkers                int32      `json:"num_workers,omitempty" tf:"group:size,conflicts:autoscale"`
```

Comment:
I ran the imperative acceptance test, my new one which specifically tests this case and also manually by using generating the binary and resizing clusters. Worked fine! 

Did you run into any issues with this before? I can definitely test any concerns you may have or just as well remove this annotation if you feel strongly about it since the code should function without it

The main motive for having a omitempty here is to make it semantically the same as the API since in the API only one of these fields can be populated at a time. Not an expert here so please let me know if it's unnecessary!

Reply:
There are serialization issues with using omitempty since the API expects the `size` field to be populated, removing these annotations.